pkg/controllers/managementuser/clusterauthtoken/token.go

```diff
@@ -153,32 +249,50 @@ func (h *tokenHandler) Updated(token *managementv3.Token) (runtime.Object, error
 }
 
 func (h *tokenHandler) Remove(token *managementv3.Token) (runtime.Object, error) {
+	return nil, h.remove(token.GetName(), token.GetUserID(), tokenUserClusterKey(token))
+}
+
+func (h *tokenHandler) remove(name, userID, key string) error {
 
-	tokens, err := h.tokenIndexer.ByIndex(tokenByUserAndClusterIndex, tokenUserClusterKey(token))
+	tokens, err := h.tokenIndexer.ByIndex(tokenByUserAndClusterIndex, key)
 	if err != nil && !errors.IsNotFound(err) {
-		return nil, err
+		return err
 	}
-	err = h.clusterAuthToken.Delete(token.Name, &metav1.DeleteOptions{})
+
+	eTokens, err := h.extTokenIndexer.ByIndex(tokenByUserAndClusterIndex, key)
```

Comment:
nit: in my opinion having a separate remove only for ext tokens make the code easier to read. Not a blocking comment, feel free to ignore and resolve if you prefer having the same `remove` for both tokens